core/src/main/scala/org/bitcoins/core/hd/BIP32Path.scala

```diff
@@ -146,6 +146,43 @@ object BIP32Path extends Factory[BIP32Path] with StringFactory[BIP32Path] {
     BIP32PathImpl(path)
   }
 
+  /** Takes in a BIP32 Path and verifies all paths are hardened
+    * @throws IllegalArgumentException is a non hardened path is found
+    */
+  def fromHardenedString(string: String): BIP32Path = {
```

Comment:
instead of redoing the implementation of `BIP32Path.fromString` you could do something much simpler:

```scala
  def fromHardenedString(string: String): BIP32Path = {
    val path = BIP32Path.fromString(string)
    require(path.forall(_.hardened), s"Did not receive all hardened paths from string, got=$string")
    path
}